src/test/scala/performanceanalysis/administrator/AdministratorActorSpec.scala

```diff
@@ -54,6 +57,25 @@ class AdministratorActorSpec(_system: ActorSystem) extends ActorSpecBase(_system
       testProbe.expectMsgPF() { case LogParserNotFound(`componentName`) => true }
     }
 
+    "respond with all registered components" in {
+      val testProbe = TestProbe("createChild4")
+      4 to 10 foreach { i =>
+        val componentName: String = s"newComponent$i"
+        testProbe.send(adminActor, RegisterComponent(componentName))
+        testProbe.expectMsg(LogParserCreated(`componentName`))
+      }
+
+      testProbe.send(adminActor, GetRegisteredComponents)
+
+      val expectedComponents:Set[String] = ((4 to 10) map (i => s"component$i")).toSet
+      testProbe.expectMsgPF() { case RegisteredComponents(actualComponents) => (expectedComponents -- actualComponents).isEmpty }
```

Comment:
I'm not sure if this line actually asserts something.

`expectedComponents should contain allOf(actualComponents)` <- using `Matchers`